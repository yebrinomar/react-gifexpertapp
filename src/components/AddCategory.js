import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {

    const [inputValue, setinputValue] = useState('');

    const handleInputChange = (e)=>{
        setinputValue(e.target.value)
    }

    const handleSubmit = (e)=>{
        e.preventDefault();
        console.log('Submit hecho');
        
        if(inputValue.trim().length > 2){
            setCategories(cats => [ inputValue, ...cats ]);
            setinputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <p> {inputValue} </p>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}             
            />
        </form>       
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

