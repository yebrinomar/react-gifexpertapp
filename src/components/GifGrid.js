//import React, { useState, useEffect } from 'react'
//import { getGifs } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

    //const [images, setImages] = useState([]);
    const {data:images, loading} = useFetchGifs(category);

    // useEffect( ()=> {
    //     getGifs(category)
    //     .then(setImages);
    // }, [category])

    return ( 
        <>
            <h3 className="card animate__animated animate__fadeIn"> {category} </h3>  
            {loading && <p className="card animate__animated animate__flash">Loading</p>}
            <div className="card-grid">

                {
                    images.map( img => {
                        return <GifGridItem  
                            key={img.id}
                            {...img}
                            //img = {img} 
                        />
                    })
                }
            </div>
        </>
       
    )
}
