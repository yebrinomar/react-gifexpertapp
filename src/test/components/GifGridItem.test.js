import React from 'react';
import {shallow} from 'enzyme';
//const { default: GifGridItem } = require("../../components/GifGridItem");
import {GifGridItem} from '../../components/GifGridItem'

describe('Prueba en el <GifGridItem />', () => {
    const title = 'Un titulo';
    const url = 'https://localhost/algo-jpg';

    let wrapper = shallow(<GifGridItem title = {title} url = {url} />);

    // beforeEach(() => {
    //     console.log('beforeEach!!!')
    //     wrapper = shallow(<GifGridItem />);
    // })

    test('debe de mostrar <GifGridItem /> correctamente', () => {
 
        expect(wrapper).toMatchSnapshot();
    
    });

    test('debe de tener un pÃ¡rrafo con el title', () => {

        const p = wrapper.find('p').text().trim();        
        expect( p ).toBe( title );
        
    }); 

    
    test('debe de tener la imagen igual al url y alt de los props', () => {

        const img = wrapper.find('img');        
        expect( img.prop('src')).toBe( url );
        expect( img.prop('alt')).toBe( title )
        
    }); 

    test('debe de tener animate__fadeIn', () => {

        const div = wrapper.find('div');  
        const className = div.prop('className'); 
        expect( className.includes('animate__fadeIn')).toBe( true );
        
    }); 
    
    // test('debe de incrementar con el boton +1', () => {
    //     const btn1 = wrapper.find('button').at(0).simulate('click');
    //     //console.log(btn1.html());
    //     const counterText = wrapper.find('h2').text().trim();
        
    //     expect( counterText ).toBe('11');
        
    // });

    // test('debe de decrementar con el boton -1', () => {
    //     const btn1 = wrapper.find('button').at(2).simulate('click');
    //     //console.log(btn1.html());
    //     const counterText = wrapper.find('h2').text().trim();
        
    //     expect( counterText ).toBe('9');
        
    // });

    // test('debe de colocar el valor por defecto con el btn reset', () => {

    //     const wrapper = shallow( <GifGridItem value = {105} /> );

    //     wrapper.find('button').at(0).simulate('click');
    //     wrapper.find('button').at(0).simulate('click');
    //     wrapper.find('button').at(1).simulate('click');
    //     //console.log(btn1.html());
    //     const counterText = wrapper.find('h2').text().trim();
        
    //     expect( counterText ).toBe('105');
        
    // })  
})
