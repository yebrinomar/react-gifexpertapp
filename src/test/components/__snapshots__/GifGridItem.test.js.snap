// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en el <GifGridItem /> debe de mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__bounce animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo-jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
